extensions/amazon-aws/src/glue.tsx

```diff
@@ -0,0 +1,291 @@
+import {
+  GlueClient,
+  ListJobsCommand,
+  GetJobRunsCommand,
+  JobRun,
+  GetJobCommand,
+  StartJobRunCommand,
+  GetJobCommandOutput,
+} from "@aws-sdk/client-glue";
+import { Action, ActionPanel, Icon, List, Image, Color, Detail, showToast, Toast } from "@raycast/api";
+import { MutatePromise, showFailureToast, useCachedPromise } from "@raycast/utils";
+import AWSProfileDropdown from "./components/searchbar/aws-profile-dropdown";
+import { isReadyToFetch, resourceToConsoleLink } from "./util";
+import { AwsAction } from "./components/common/action";
+
+interface GlueJobRun extends JobRun {
+  accessoriesText?: string;
+  icon?: Icon;
+  iconTintColor?: Color;
+}
+
+export default function Glue() {
+  const { data: glueJobs, error, isLoading, mutate } = useCachedPromise(fetchJobs);
+  const {
+    data: glueJobRuns,
+    isLoading: isLoadingJobRuns,
+    mutate: mutateJobRuns,
+  } = useCachedPromise(fetchJobRuns, [1, glueJobs]);
```

Comment:
I am going to implement it. Need to figure out which improvement you exactly mean

Reply:
For sure take a look at _hooks/use-ddb.ts_ or _hooks/use-pipelines.ts_ from this PR: https://github.com/raycast/extensions/pull/13667. Essentially, 2 things:
1. There is always a call to list the pipelines/tables (or in your case jobs, `ListGlueJobs`). You can specify `MaxItems` (or corresponding limit parameter) to slow down the TPS for parallel requests in the next step.
2. After we have received the page of Jobs/Pipelines/TableNames, we make async calls to get more granular details for the job in parallel: `ListGlueJobRuns` in your case. Right now, your `ListGlueJobRuns` is sequential. I pipelines example, we made the call to `GetPipelineExecutions` similary.
3. If there are more pages for the jobs/pipelines, we repeat the process and aggregate the result.

The thing to balance is speed/latency vs TPS as explained in the pipelines issue. We don't want to make too many requests to get throttled and if we make sequential requests it might be too slow.